Lesson12/Lesson12_2.cpp

```diff
@@ -0,0 +1,103 @@
+// Lesson 12 2
+#include <iostream>
+
+enum class SortingDirection {ByRows, ByColumn};
+
+const int ROWS = 5;
+const int COLUMNS = 10;
+
+int partition(int array[], int low, int high)
+{
+    int pivot = array[high];
+
+    // idx of greater element
+    int pointerIdx = (low - 1);
+
+    for (int j = low; j < high; j++)
+    {
+        if (array[j] <= pivot)
+        {
+            // if element smaller than pivot is found
+            // swap it with the greater element pointed by i
+            pointerIdx++;
+
+            // swap element at i with element at j
+            std::swap(array[pointerIdx], array[j]);
+        }
+    }
+
+    std::swap(array[pointerIdx + 1], array[high]);
+
+    return (pointerIdx + 1);
+}
+
+void quickSort(int array[], int low, int high)
+{
+    if (low < high)
+    {
+        int partition_idx = partition(array, low, high);
+        quickSort(array, low, partition_idx - 1);
+        quickSort(array, partition_idx + 1, high);
+    }
+}
+
+void sort(int arr[ROWS][COLUMNS], SortingDirection sd)
+{
+    int arrByRow[ROWS];
+    int arrByCol[COLUMNS];
+    if(sd == SortingDirection::ByRows){
+        for(int i = 0; i < COLUMNS; i++) {
+            for(int j = 0; j < ROWS; j++) {
+                arrByRow[j] = arr[j][i];
+            }
+            quickSort(arrByRow, 0, ROWS);
+
+            for(int k = 0; k < ROWS; k++) {
+                arr[k][i] = arrByRow[k];
+            }
+        }
+        std::cout << "Sorted by rows array: " << std::endl;
+    } else {
+        for(int i = 0; i < ROWS; i++){
+            for(int j = 0; j < COLUMNS; j++) {
+                arrByCol[j] = arr[i][j];
+            }
+            quickSort(arrByCol, 0, COLUMNS);
```

Comment:
you don't need additional array here. you can call sort like this:
`quickSort(arr[i], 0, COLUMNS);`

Reply:
fixed